### PR TITLE
Moves and renames

### PR DESCRIPTION
**Pull request by Jozkee**
Fixes https://github.com/dotnet/command-line-api/issues/1900
Fixes https://github.com/dotnet/command-line-api/issues/1899
Fixes https://github.com/dotnet/command-line-api/issues/1898
Fixes https://github.com/dotnet/command-line-api/issues/1897
Fixes https://github.com/dotnet/command-line-api/issues/1896

### CODE REVIEW

File: src/System.CommandLine.ApiCompatibility.Tests/ApiCompatibilityApprovalTests.System_CommandLine_api_is_not_changed.approved.txt [hunk 1]
```diff
@@ -32,19 +40,11 @@ System.CommandLine
...
-    public static Argument<System.IO.FileInfo> ExistingOnly(this Argument<System.IO.FileInfo> argument)
-    public static Argument<System.IO.DirectoryInfo> ExistingOnly(this Argument<System.IO.DirectoryInfo> argument)
-    public static Argument<System.IO.FileSystemInfo> ExistingOnly(this Argument<System.IO.FileSystemInfo> argument)
-    public static Argument<T> ExistingOnly<T>(this Argument<T> argument)
-    public static TArgument FromAmong<TArgument>(this TArgument argument, System.String[] values)
-    public static TArgument LegalFileNamesOnly<TArgument>(this TArgument argument)
-    public static TArgument LegalFilePathsOnly<TArgument>(this TArgument argument)
-    public static ParseResult Parse(this Argument argument, System.String commandLine)
-    public static ParseResult Parse(this Argument argument, System.String[] args)
+  public static class ArgumentValidationExtensions
```
Comment (jonsequitur) [inline src/System.CommandLine.ApiCompatibility.Tests/ApiCompatibilityApprovalTests.System_CommandLine_api_is_not_changed.approved.txt:43]: Would `ArgumentValidation` be a more standard class name?

Comment (Jozkee) [reply src/System.CommandLine.ApiCompatibility.Tests/ApiCompatibilityApprovalTests.System_CommandLine_api_is_not_changed.approved.txt:43]: @bartonjs may know better, but I think is a common pattern to use `Extensions` suffix for extension classes.

Comment (bartonjs) [reply src/System.CommandLine.ApiCompatibility.Tests/ApiCompatibilityApprovalTests.System_CommandLine_api_is_not_changed.approved.txt:43]: Adding the Extension suffix is common, but is actually an anti-pattern.  You should, ideally, name the containing type on the basis everything in it gets invoked as a static at the callsite, then can optionally make things be extension methods.

Also, extension methods themselves are generally an anti-pattern.

* If the type is an interface, see the design guidelines that basically suggest "don't add new interfaces, they're always full of regret".
* If the type isn't generic, you shouldn't have any extension methods for it in the same assembly.
* If the type is generic, and you have members that only apply for a smaller subset of Ts, those can/have-to be generic.
* If the type is generic, expects to be instantiated with nearly value-type under the sun, and needs to be small (e.g. it's Span) then extension methods tend to be preferred so that the T-substitutions are reduced to only the combination of T and method that got used.

So, basically, System.CommandLine should have almost no extension methods.

Comment (Jozkee) [reply src/System.CommandLine.ApiCompatibility.Tests/ApiCompatibilityApprovalTests.System_CommandLine_api_is_not_changed.approved.txt:43]: > If the type is generic, and you have members that only apply for a smaller subset of Ts, those can/have-to be generic.

That's the case here. 

I prefer the `Extensions` suffix just because it glimpses that is a class meant to contain extensions but is not a strong opinion. Thanks, @bartonjs.

**Review by jonsequitur — APPROVED** [review]
